plugins/workspace-minimap/test/minimap_tests.mocha.js

```diff
@@ -0,0 +1,332 @@
+            y: this.minimapMetrics.svgHeight};
+          const converted = Minimap.minimapToPrimaryCoords(
+              this.primaryMetrics, this.minimapMetrics,
+              click.x, click.y);
+
+          assert.deepEqual(converted, [750, 0],
+              'Incorrect bottom right click');
+        });
+      });
+
+      suite('Tall, small, and bottom right shifted content', function() {
+        setup(function() {
+          this.primaryMetrics = {
+            contentHeight: 2000, contentWidth: 500,
+            contentTop: 500, contentLeft: 500,
+            viewWidth: 500, viewHeight: 500};
+          this.minimapMetrics = {svgWidth: 500, svgHeight: 500,
+            contentHeight: 500, contentWidth: 125};
+        });
+
+        test('Top left click', function() {
+          const click = {x: 0, y: 0};
+          const converted = Minimap.minimapToPrimaryCoords(
+              this.primaryMetrics, this.minimapMetrics,
+              click.x, click.y);
+
+          assert.deepEqual(converted, [500, -250],
+              'Incorrect top left click');
+        });
+
+        test('Center click', function() {
+          const click = {
+            x: this.minimapMetrics.svgWidth / 2,
+            y: this.minimapMetrics.svgHeight / 2};
+          const converted = Minimap.minimapToPrimaryCoords(
+              this.primaryMetrics, this.minimapMetrics,
+              click.x, click.y);
+
+          assert.deepEqual(converted, [-500, -1250],
+              'Incorrect center click');
+        });
+
+        test('Bottom right click', function() {
+          const click = {
+            x: this.minimapMetrics.svgWidth,
+            y: this.minimapMetrics.svgHeight};
+          const converted = Minimap.minimapToPrimaryCoords(
+              this.primaryMetrics, this.minimapMetrics,
+              click.x, click.y);
+
+          assert.deepEqual(converted, [-1500, -2250],
+              'Incorrect bottom right click');
+        });
+      });
+    });
+
+
+suite('Positioning the minimap in the primary workspace', function() {
+  setup(function() {
+    this.mockMetrics = {
+      viewMetrics: {
+        height: 1000,
+        width: 1000,
+      },
+      absoluteMetrics: {
+        left: 0,
+        top: 0,
+      },
+      toolboxMetrics: {
+        position: Blockly.TOOLBOX_AT_LEFT,
+      },
+    };
+
+    this.options = {
+      RTL: true,
+      horizontalLayout: true,
+      toolboxPosition: 'start',
+      move: {scrollbars: true},
+      scrollbar: {
+        isVisible() {
+          return true;
+        },
+        canScrollVertically() {
+          return true;
+        },
+        canScrollHorizontally() {
+          return true;
+        },
+      },
+      getMetrics() {
+        return {viewWidth: 100};
+      },
+    };
+  });
+
+  test('LTR Vertical Start', function() {
+    this.options.RTL = false;
+    this.options.horizontalLayout = false;
+    this.options.toolboxPosition = 'start';
```

Comment:
Throughout, WDYT about doing 

```
Object.assign(this.options, {
  RTL: false,
  horizontalLayout: false,
  toolboxPosition: start
});
```

I feel like that might make the fields that are being changed a bit clearer by removing the this.options. boilerplate?

Reply:
Love that